src/main/java/com/minecolonies/coremod/util/InventoryFunctions.java

```diff
@@ -78,51 +78,56 @@ public static boolean matchFirstInInventoryWithInventory(
+    private static boolean matchFirstInProvider(
+                                                  final ICapabilityProvider provider, @NotNull final Function<ICapabilityProvider, Function<Integer,
                                                                                                                             Predicate<ItemStack>>> tester)
     {
-        return matchInInventory(inventory, tester, true);
+        return matchInProvider(provider, tester, true);
     }
 
     /**
-     * Topmost function to actually loop over the inventory.
+     * Topmost function to actually loop over the provider.
      * Will return if it found something.
      *
-     * @param inventory      the inventory to loop over
+     * @param provider      the provider to loop over
      * @param tester         the function to use for testing slots
      * @param stopAfterFirst if it should stop executing after finding one stack that applies
      * @return true if it found a stack
      */
-    private static boolean matchInInventory(
-                                             @Nullable final IInventory inventory,
-                                             @NotNull final Function<IInventory, Function<Integer,
+    private static boolean matchInProvider(
+                                             @Nullable final ICapabilityProvider provider,
+                                             @NotNull final Function<ICapabilityProvider, Function<Integer,
                                                                                            Predicate<ItemStack>>> tester,
                                              final boolean stopAfterFirst)
     {
-        if (inventory == null)
+        if (provider == null)
         {
             return false;
         }
-        final int size = inventory.getSizeInventory();
+
         boolean foundOne = false;
-        for (int slot = 0; slot < size; slot++)
-        {
-            final ItemStack stack = inventory.getStackInSlot(slot);
-            //Unchain the function and apply it
-            if (tester.apply(inventory).apply(slot).test(stack))
+        for(IItemHandler handler : InventoryUtils.getItemHandlersFromProvider(provider)) {
+            final int size = handler.getSlots();
+            for (int slot = 0; slot < size; slot++)
             {
-                foundOne = true;
-                if (stopAfterFirst)
+                final ItemStack stack = handler.getStackInSlot(slot);
+                //Unchain the function and apply it
+                if (tester.apply(provider).apply(slot).test(stack))
                 {
-                    return true;
+                    foundOne = true;
+                    if (stopAfterFirst)
+                    {
+                        return true;
+                    }
                 }
             }
+            return foundOne;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Avoid using a branching statement as the last in a loop. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3AAvoidBranchingStatementAsLastInLoop)